### ISSUE TITLE
Officially deprecate `Dropdown`

### ISSUE THREAD

**Issue by khiga8** [primer/view_components#2062]
It appears `Dropdown` should be deprecated in favor of `ActionMenu`. This could be a great candidate for a component migration linting rule, which could be backed by the accessibility scorecard.

**Comment by jonrohan** [comment]
We're tracking this here https://github.com/github/primer/issues/2044
